UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Controllers/ProductDataControllerTests.cs

```diff
@@ -0,0 +1,113 @@
+
+        {
+            var validationMessage = new ValidationFailure("RequestBody", "Either body is null or malformed.");
+            validationMessage.ErrorCode = HttpStatusCode.BadRequest.ToString();
+
+            A.CallTo(() => fakeProductDataService.ValidateProductDataByProductVersions(A<ProductDataProductVersionsRequest>.Ignored))
+                            .Returns(new ValidationResult(new List<ValidationFailure> { validationMessage }));
+
+            var result = (BadRequestObjectResult)await controller.PostProductDataByProductVersions(new List<ProductVersionRequest>(), "");
+            var errors = (ErrorDescription)result.Value;
+            Assert.AreEqual(400, result.StatusCode);
+            Assert.AreEqual("Either body is null or malformed.", errors.Errors.Single().Description);
+        }
+
+        [Test]
+        public async Task WhenValidProductVersionRequest_ThenPostProductDataByProductVersionsReturnsOkResponse()
+        {
+            var response = new ExchangeSetResponse();
+            response.Links = new Links();
+            response.Links.ExchangeSetBatchStatusUri = new LinkSetBatchStatusUri()
+            {
+                Href = "http://fss.ukho.gov.uk/batch/7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272"
+            };
+            response.Links.ExchangeSetFileUri = new LinkSetFileUri()
+            {
+                Href = "http://fss.ukho.gov.uk/batch/7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272/files/exchangeset123.zip"
+            };
+            response.ExchangeSetUrlExpiryDateTime = DateTime.Now;
+            response.RequestedProductCount = 22;
+            response.ExchangeSetCellCount = 15;
+            response.RequestedProductsAlreadyUpToDateCount = 5;
+            response.RequestedProductsNotInExchangeSet = new List<RequestedProductsNotInExchangeSet>() {
+                new RequestedProductsNotInExchangeSet()
+                {
+                    ProductName = "GB123456",
+                    Reason = "productWithdrawn"
+                },
+                new RequestedProductsNotInExchangeSet()
+                {
+                    ProductName = "GB123789",
+                    Reason = "invalidProduct"
+                }
+            };
+            A.CallTo(() => fakeProductDataService.ValidateProductDataByProductVersions(A<ProductDataProductVersionsRequest>.Ignored))
+                            .Returns(new ValidationResult(new List<ValidationFailure>()));
+
+            A.CallTo(() => fakeProductDataService.CreateProductDataByProductVersions(A<ProductDataProductVersionsRequest>.Ignored))
+                 .Returns(new ExchangeSetResponse());
+
+            var result = (OkObjectResult)await controller.PostProductDataByProductVersions(new List<ProductVersionRequest>() 
+                            { new ProductVersionRequest() { ProductName = "demo" } }, "");
+            Assert.IsInstanceOf<OkObjectResult>(result);
```

Comment:
can you change this to be an Assert AreSame(response,result.Value) - not sure if this will compile
```suggestion
            Assert.AreSame(response,result.Value);
```

Reply:
used Assert.AreSame instead of Assert.IsInstanceOf